editor.planx.uk/src/pages/FlowEditor/lib/store/user.ts

```diff
@@ -0,0 +1,48 @@
+import { Team, User } from "types";
+import type { StateCreator } from "zustand";
+
+export interface UserStore {
+  name: string;
+  email: string;
+  isPlatformAdmin: boolean;
+  roles: { teamSlug: Team["slug"]; role: "teamEditor" | "teamViewer" }[] | [];
+
+  setUser: (user: User) => void;
+  getUser: () => User;
+  canUserEditTeam: (teamSlug: Team["slug"]) => boolean;
+}
+
+export const userStore: StateCreator<UserStore, [], [], UserStore> = (
+  set,
+  get,
+) => ({
+  name: "",
+  email: "",
+  isPlatformAdmin: false,
+  roles: [],
+
+  setUser: (user) =>
+    set({
+      name: user.name,
+      email: user.email,
+      isPlatformAdmin: user.isPlatformAdmin,
+      roles: user.roles,
+    }),
+
+  getUser: () => ({
+    name: get().name,
+    email: get().email,
+    roles: get().roles,
+    isPlatformAdmin: get().isPlatformAdmin,
+  }),
+
+  canUserEditTeam: (teamSlug) => {
+    return (
+      get().roles.filter(
+        (r) =>
+          (r.role === "teamEditor" && r.teamSlug === teamSlug) ||
+          get().isPlatformAdmin,
+      ).length > 0
+    );
+  },
```

Comment:
this is the most important bit of logic to doublecheck please ! 

intended behavior here is: 
- return `true` if I'm a `platformAdmin` independent of `teamSlug`
- return `true` if I'm a `teamEditor` of the given `teamSlug`
- return `false` otherwise

Reply:
The logic here is correct 👍